src/Compilers/CSharp/Portable/Binder/Binder_Operators.cs

```diff
@@ -2342,10 +2348,28 @@ private bool CheckConstraintLanguageVersionAndRuntimeSupportForOperator(SyntaxNo
                 }
             }
 
-            if (methodOpt is not null && SyntaxFacts.IsCheckedOperator(methodOpt.Name) &&
-                Compilation.SourceModule != methodOpt.ContainingModule)
+            if (methodOpt is null)
+            {
+                if (isUnsignedRightShift)
+                {
+                    result &= CheckFeatureAvailability(node, MessageID.IDS_FeatureUnsignedRightShift, diagnostics);
+                }
+            }
+            else
             {
-                result &= CheckFeatureAvailability(node, MessageID.IDS_FeatureCheckedUserDefinedOperators, diagnostics);
+                Debug.Assert((methodOpt.Name == WellKnownMemberNames.UnsignedRightShiftOperatorName) == isUnsignedRightShift);
+
+                if (Compilation.SourceModule != methodOpt.ContainingModule)
```

Comment:
I didn't understand why the unsigned right shift operator LangVer check is conditioned on resolved method being in source. I would have expected an unconditional LangVer diagnostic for `>>>` where ever the best signature is found.

Reply:
>I didn't understand why the unsigned right shift operator LangVer check is conditioned on resolved method being in source. I would have expected an unconditional LangVer diagnostic for ```>>>``` where ever the best signature is found.

1. This logic is not new
2. If the operator is declared in the same module (note, not just in source), we perform the check on the declaration and report the error, if needed. Reporting the same error at every consumption site would only add more noise.